smartsim/settings/pbsSettings.py

```diff
@@ -189,8 +192,22 @@ def _create_resource_list(self) -> t.List[str]:
         res = []
 
         # get select statement from resources or kwargs
-        if "select" in self.resources:
+        if ("select" in self.resources) and "nodes" not in self.resources:
             res += [f"-l select={str(self.resources['select'])}"]
+        elif ("select" in self.resources) and ("nodes" in self.resources):
+            nselect = self.resources["select"]
+            if nselect == self._nodes:
```

Comment:
Fixed. We now check in the Constructor to ensure that the `nodes` kwarg and `nodes` key in `resources` are either the same value. Otherwise we throw an error

Reply:
This corner case is now handled and tested. The type error here no longer exists as this clause was removed